src/plugins/action_trim_responses/src/trimResponses.ts

```diff
@@ -1,63 +1,77 @@
 import databaseMethods from './databaseMethods'
-import isEqual from '@openmsupply/deep-comparison'
-
+const isEqual = require('deep-equal')
 interface Response {
   id: number
   code: string
+  template_element_id?: number
+  application_response_id?: number
+  review_response_link_id?: number
   value: { [key: string]: any }
   comment?: string
   decision?: string
-  time_created: any
+  time_updated: any
 }
 
-interface ResponsesByCode {
-  [key: string]: Response[]
+interface ResponsesById {
+  [key: number]: Response[]
 }
 
 module.exports['trimResponses'] = async function (input: any, DBConnect: any) {
   const db = databaseMethods(DBConnect)
 
-  console.log('Trimming unchanged duplicated responses...')
+  const { applicationId, reviewId, timestamp } = input
+  console.log(`Trimming unchanged duplicated ${reviewId ? 'Review' : 'Application'} responses...`)
 
   try {
-    const { applicationId, reviewId } = input
     // Get ALL responses associated with application OR review
     const responses = reviewId
       ? await db.getAllReviewResponses(reviewId)
       : await db.getAllApplicationResponses(applicationId)
 
-    // Create object of responses indexed by code, with response
-    // objects added to array for each code
-    const responsesByCode: ResponsesByCode = {}
-    responses.forEach((response: Response) => {
-      const { code } = response
-      if (reviewId) response.value = { comment: response.comment, decision: response.decision }
-      if (!(code in responsesByCode)) responsesByCode[code] = [response]
-      else responsesByCode[code].push(response)
-    })
+    const reviewLevel = responses?.[0]?.level
+
+    // Create object of indexed responses, with array of response
+    // objects for each indexed id
 
+    const groupByField = !reviewId
+      ? 'template_element_id'
```

Comment:
this is stylistic (so not required change, I've suggested this to Nicole a few times), but more nested in-line condition statement are much harder to read then just if statement.

Reply:
Yeah, I agree in general, although when the expressions are short I quite like the way Prettier formats them to be quite logical. But yeah, definitely when the expressions are long -- like chunks of JSX! ;)